apps/ecommerce/frontend/src/components/Common/ProductCard/ProductCard/ProductCard.tsx

```diff
@@ -0,0 +1,104 @@
+import { useState } from 'react';
+import { Card } from '@contentful/f36-components';
+import { styles } from './ProductCard.styles';
+import { ExternalResource, ExternalResourceError, ProductCardType } from 'types';
+import ProductCardHeader from '../ProductCardHeader/ProductCardHeader';
+import ProductCardBody from '../ProductCardBody/ProductCardBody';
+import { RenderDragFn } from '@contentful/field-editor-reference/dist/types';
+import ProductCardRawData from '../ProductCardRawData/ProductCardRawData';
+
+type CardMovementCallbacks = {
+  handleMoveToTop?: (index?: number) => void;
+  handleMoveToBottom?: (index?: number) => void;
+};
+
+export interface ProductCardProps {
+  // TODO: Fix the CardHeader type during the mapping config refactor/ticket
+  resource: any;
+  cardHeader: string;
+  productCardType?: ProductCardType;
+  handleRemove?: (index?: number) => void;
+  onSelect?: (resource: ExternalResource) => void;
+  isSelected?: boolean;
+  dragHandleRender?: RenderDragFn;
+  isLoading?: boolean;
+  cardIndex?: number;
+  totalCards?: number;
+  externalResourceLink?: any;
+  cardMovementCallbacks?: CardMovementCallbacks;
+  isHovered?: boolean;
+  externalResourceError?: ExternalResourceError;
+}
+
+const ProductCard = (props: ProductCardProps) => {
+  const [showJson, setShowJson] = useState<boolean>(false);
+
+  const {
+    cardIndex,
+    totalCards,
+    productCardType = 'dialog',
+    isLoading,
+    resource,
+    cardHeader,
+    handleRemove,
+    onSelect,
+    isSelected,
+    dragHandleRender,
+    externalResourceLink,
+    cardMovementCallbacks,
+    isHovered,
+    externalResourceError,
+  } = props;
+
+  const { handleMoveToBottom, handleMoveToTop } = cardMovementCallbacks || {};
+  const fieldProductCardType = productCardType === 'field';
+  const renderRawData = !externalResourceError?.error && fieldProductCardType && externalResourceLink && showJson;
+
+  return (
+    <Card
+      padding="none"
+      className={styles.productCard}
+      isSelected={isSelected}
+      onClick={() => {
+        onSelect && onSelect(resource);
+      }}
+      isLoading={isLoading}
+      withDragHandle={!!dragHandleRender}
+      dragHandleRender={dragHandleRender}
+      // TODO: Determine hover state prop
+      isHovered={isHovered}>
+      <ProductCardHeader
+        headerTitle={cardHeader}
+        status={resource.status}
+        handleRemove={handleRemove}
+        cardIndex={cardIndex}
+        totalCards={totalCards}
+        showJson={showJson}
+        handleShowJson={setShowJson}
+        showExternalResourceLinkDetails={!externalResourceError?.error && Boolean(externalResourceLink)} // do we want to show this icon button when there is an error, or the showData option?
```

Comment:
question about showing the external link icon - as well as conditionally rendering the 'Show data' menu item if there exists an error!

Reply:
Yeah totally agree - sorry my wording was a bit confusing, I meant do we want this to show if an error exists? The default state is that the icon shows, but the conditional here in the code is that it only shows if no error exists